Subject: [PATCH] adapt modify in uos
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision 96a320f812f79e928894e93dada536b442a32d32)
+++ b/CMakeLists.txt	(revision 937c8250f13b0a463be6ca06d05d10b7d17bf216)
@@ -22,12 +22,22 @@
 endif()
 
 # you should config right path according to your run-time environment
-set(KPATH /usr/src/kernels)
-execute_process(COMMAND ls ${KPATH} OUTPUT_VARIABLE KDIR)
+set(KPATH /usr/src)
+execute_process(COMMAND ls ${KPATH} | grep ${COMMAND uname -r} OUTPUT_VARIABLE KDIR)
 # Remove the last char \n
 string(STRIP ${KDIR} KDIR)
 set(KERNEL_DIR ${KPATH}/${KDIR})
 
+# Add header file include directories
+include_directories(
+    ./
+    ./libboundscheck/include
+    ./auth
+    ./core
+    ./tlogger
+    ./kthread_affinity
+)
+
 # Set extra options
 set(CMAKE_EXTRA_FLAGS "-fstack-protector-strong -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_CLOUDSERVER_TEECD_AUTH")
 set(CMAKE_EXTRA_FLAGS "${CMAKE_EXTRA_FLAGS} -DCONFIG_CPU_AFF_NR=0 -DCONFIG_BIG_SESSION=1000 -DCONFIG_NOTIFY_PAGE_ORDER=4 -DCONFIG_512K_LOG_PAGES_MEM -DCONFIG_TEE_TRACE")
Index: Makefile
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Makefile b/Makefile
--- a/Makefile	(revision 96a320f812f79e928894e93dada536b442a32d32)
+++ b/Makefile	(revision 937c8250f13b0a463be6ca06d05d10b7d17bf216)
@@ -18,8 +18,8 @@
 endif
 
 # you should config right path according to your run-time environment
-KPATH := /usr/src/kernels
-KDIR  := $(KPATH)/$(shell ls $(KPATH))
+KPATH := /usr/src
+KDIR  := $(KPATH)/$(shell dpkg -l | grep linux-headers | awk '{print $$2}' | head -n 1)
 
 EXTRA_CFLAGS += -fstack-protector-strong -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_CLOUDSERVER_TEECD_AUTH
 EXTRA_CFLAGS += -I$(PWD)/libboundscheck/include/ -I$(PWD) -I$(PWD)/auth -I$(PWD)/core -I$(PWD)/tzdriver_internal/tee_trace_event
Index: README.en.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/README.en.md b/README.en.md
--- a/README.en.md	(revision 96a320f812f79e928894e93dada536b442a32d32)
+++ b/README.en.md	(revision 937c8250f13b0a463be6ca06d05d10b7d17bf216)
@@ -28,6 +28,7 @@
 2) # insmod tzdriver.ko
 3) # /usr/bin/teecd &
 4) run any CA
+5) If you want to detele tzdriver after installing it, you need to reboot the server.
 
 5.License
 please see License/Tzdriver_License for more details
Index: README.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/README.md b/README.md
--- a/README.md	(revision 96a320f812f79e928894e93dada536b442a32d32)
+++ b/README.md	(revision 937c8250f13b0a463be6ca06d05d10b7d17bf216)
@@ -25,3 +25,4 @@
    2）使用root用户，执行insmod tzdriver.ko
    3）使用root用户，执行/usr/bin/teecd&
    4）运行测试CA 和TA
+   5）如果要卸载tzdriver，需要重启服务器才可生效
Index: tlogger/tlogger.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tlogger/tlogger.c b/tlogger/tlogger.c
--- a/tlogger/tlogger.c	(revision 96a320f812f79e928894e93dada536b442a32d32)
+++ b/tlogger/tlogger.c	(revision 937c8250f13b0a463be6ca06d05d10b7d17bf216)
@@ -645,7 +645,7 @@
 static int check_user_arg(unsigned long arg, size_t arg_len)
 {
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 18) || \
-	LINUX_VERSION_CODE == KERNEL_VERSION(4, 19, 71))
+	LINUX_VERSION_CODE == KERNEL_VERSION(4, 19, 71) || UOS == 1)
 	return (int)access_ok(VERIFY_READ,
 		(void __user *)(uintptr_t)arg, arg_len);
 #else
